packages/babel-parser/src/plugins/typescript/index.ts

```diff
@@ -3165,21 +3125,93 @@ export default (superClass: ClassWithMixin<typeof Parser, IJSXParserMixin>) =>
       flags: ParseStatementFlag,
       decorators?: N.Decorator[] | null,
     ): N.Statement {
-      if (this.match(tt._const) && this.isLookaheadContextual("enum")) {
-        const node = this.startNode<N.TsEnumDeclaration>();
-        this.expect(tt._const); // eat 'const'
-        return this.tsParseEnumDeclaration(node, { const: true });
-      }
-
-      if (this.isContextual(tt._enum)) {
-        return this.tsParseEnumDeclaration(
-          this.startNode<N.TsEnumDeclaration>(),
-        );
-      }
-
-      if (this.isContextual(tt._interface)) {
-        const result = this.tsParseInterfaceDeclaration(this.startNode());
-        if (result) return result;
+      if (!this.state.containsEsc) {
```

Comment:
This check is redundant for `tt._const`, however it is more convenient to group `tt._const` handling with other keyword-likes.